front/pages/w/[wId]/u/gens/index.tsx

```diff
@@ -1625,17 +1548,126 @@ export default function AppGens({
+                      {dataSources.map((ds) => {
+                        return (
+                          <div
+                            key={ds.name}
+                            className="group ml-1 flex flex-initial"
+                          >
+                            <div
+                              className={classNames(
+                                "z-10 flex h-4 w-4 flex-initial cursor-pointer",
+                                ds.provider !== "none" ? "mr-1" : "",
+                                ds.selected ? "opacity-100" : "opacity-25"
+                              )}
+                              onClick={() => {
+                                handleSwitchDataSourceSelection(ds.name);
+                              }}
+                            >
+                              {ds.provider !== "none" ? (
+                                <img
+                                  src={PROVIDER_LOGO_PATH[ds.provider]}
+                                ></img>
+                              ) : (
+                                <DocumentDuplicateIcon className="-ml-0.5 h-4 w-4 text-slate-500" />
+                              )}
+                            </div>
+                            <div className="absolute z-0 hidden rounded leading-3 group-hover:block">
+                              <div className="relative bottom-8 border bg-white px-1 py-1 ">
+                                <span className="text-gray-600">
+                                  <span className="font-semibold">
+                                    {ds.name}
+                                  </span>
+                                  {ds.description ? ` ${ds.description}` : null}
+                                </span>
+                              </div>
+                            </div>
+                          </div>
+                        );
+                      })}
+                    </div>
+                  </div>
+                  <div className="flex flex-row items-center space-x-2 leading-8">
+                    <div className="flex flex-initial text-gray-400">
+                      Time range:
+                    </div>
+                    <div className="flex flex-initial cursor-pointer text-gray-400">
+                      {inferTimeRangeLoading ? (
+                        <div className="mt-0.5">
+                          <Spinner />
+                        </div>
+                      ) : (
+                        <SparklesIcon
+                          className="h-4 w-4 text-yellow-400"
+                          onClick={handleInferTimeRange}
+                        />
+                      )}
+                    </div>
+                    <GensTimeRangePicker
+                      timeRange={timeRange}
+                      onTimeRangeUpdate={setTimeRange}
+                    />
+                  </div>
+                  <div className="flex flex-row items-center space-x-2 leading-8">
```

Comment:
This won't work aligned on the same line if you click on the time picker? Let's keep them one per line for now?